### PR TITLE
feat: implement `CopyToClipboardButton` component on `@rainbow-modules/record` module

### PR DESCRIPTION
**Pull request by yvmunayev**
#235

### CODE REVIEW

File: packages/record/src/components/CopyToClipboardButton/index.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    );
+}
+
+CopyToClipboardButton.propTypes = {
+    /** Text to be copied to clipboard */
+    value: PropTypes.string,
+    /** Text for the tooltip that describes the action of copying to clipboard. */
+    copyText: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+    /** text for the tooltip that describes that the text was copied to the clipboard. */
+    copiedText: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
```
Comment (LeandroTorresSicilia) [inline packages/record/src/components/CopyToClipboardButton/index.js:102]: these texts will be translated by keys

File: packages/record/src/components/CopyToClipboardButton/index.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        triggerElementRef: () => triggerRef.current.htmlElementRef,
+    });
+
+    if (!isSupported()) {
+        return null;
+    }
+
+    const handleClick = () => {
+        const success = copy(value);
+        triggerRef.current.focus();
```
Comment (LeandroTorresSicilia) [inline packages/record/src/components/CopyToClipboardButton/index.js:52]: why need to focus manually?

File: packages/record/src/components/CopyToClipboardButton/index.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        if (success) {
+            setCopied(true);
+            setVisible(true);
+        }
+    };
+
+    const hiddenTooltip = () => {
```
Comment (LeandroTorresSicilia) [inline packages/record/src/components/CopyToClipboardButton/index.js:59]: this is an action then we should call it `hideTooltip`

File: packages/record/src/components/CopyToClipboardButton/helpers/__test__/isSupported.spec.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+describe('<isSupported/>', () => {
+    it('should return true when window.clipboardData and window.clipboardData.setData is defined', () => {
+        window.clipboardData = {
+            setData: jest.fn(),
+        };
+        expect(isSupported()).toBeTruthy();
+    });
+    it('should return true when document.queryCommandSupported is defined and document.queryCommandSupported("copy") return true', () => {
+        document.queryCommandSupported = jest.fn(() => true);
+        expect(isSupported()).toBeTruthy();
```
Comment (LeandroTorresSicilia) [inline packages/record/src/components/CopyToClipboardButton/helpers/__test__/isSupported.spec.js:12]: here window.clipboardData.setData does not exists by default?

Comment (yvmunayev) [reply packages/record/src/components/CopyToClipboardButton/helpers/__test__/isSupported.spec.js:12]: window.clipboardData.setData is an Internet Explore API

File: packages/record/src/components/CopyToClipboardButton/index.js (hunk 2 above)
Comment (yvmunayev) [reply packages/record/src/components/CopyToClipboardButton/index.js:52]: In the copy function when doing this:

``` js
const range = document.createRange ();
const selection = document.getSelection ();

range.selectNodeContents (mark);
selection.addRange (range);
```

the text is selected to copy it, the focus moves to this element so it must be changed manually.

File: packages/record/src/components/CopyToClipboardButton/index.js [hunk 5]
```diff
@@ -26,7 +25,9 @@ export default function CopyToClipboardButton(props) {
     const tooltipRef = useRef();
     const [isVisible, setVisible] = useState(false);
     const [isCopied, setCopied] = useState(false);
-    const text = isCopied ? copiedText : copyText;
+    const copyProps = { id: 'click_to_copy', defaultMessage: 'Click to copy' };
+    const copiedProps = { id: 'copied', defaultMessage: 'Copied' };
```
Comment (LeandroTorresSicilia) [inline packages/record/src/components/CopyToClipboardButton/index.js:29]: look at other components like `packages/payment/src/components/CreditCardPicker/messages.js`
where we define a messages file and then set an id following a format to and then document these ids in order to let the user use them to translate with their custom dictionarie

**Review by LeandroTorresSicilia — APPROVED** [review]

**Pull request merged**